kernel/sched.c

```diff
@@ -1258,7 +1258,7 @@ extern void thread_abort_hook(struct k_thread *thread);
  * @param thread Identify the thread to halt
  * @param new_state New thread state (_THREAD_DEAD or _THREAD_SUSPENDED)
  */
-static void halt_thread(struct k_thread *thread, uint8_t new_state)
+static ALWAYS_INLINE void halt_thread(struct k_thread *thread, uint8_t new_state)
```

Comment:
I think I count three spots where this medium-sized function is called, so an ALWAYS_INLINE is going to have a non-trivial code size impact.  Did you look at how bad it is?  My gut says that this is too big to be inlining, but will defer to numbers.

Reply:
Inlining this code showed a 124 byte code size increase on disco_l475_iot1 and a 7.8% performance boost. On qemu_x86, the code size increased by 128 bytes.